lib/SILGen/SILGenStmt.cpp

```diff
@@ -1248,11 +1248,45 @@ void StmtEmitter::visitRepeatWhileStmt(RepeatWhileStmt *S) {
 void StmtEmitter::visitForEachStmt(ForEachStmt *S) {
   // Emit the 'iterator' variable that we'll be using for iteration.
   LexicalScope OuterForScope(SGF, CleanupLocation(S));
-  {
-    SGF.emitPatternBinding(S->getIteratorVar(),
-                           /*index=*/0, /*debuginfo*/ true);
+
+  if (auto *expansion =
+          dyn_cast<PackExpansionExpr>(S->getTypeCheckedSequence())) {
+    auto formalPackType = dyn_cast<PackType>(
+        PackType::get(SGF.getASTContext(), expansion->getType())
+            ->getCanonicalType());
+
+    JumpDest loopDest = createJumpDest(S->getBody());
+
+    // Set the destinations for 'break' and 'continue'.
+    JumpDest endDest = createJumpDest(S->getBody());
```

Comment:
I think `loopDest` and `endDest` would be far clearer as `continueDest` and `breakDest` for anyone yet to read this. Just an opinion.

Reply:
I just saw this. I think this makes a lot of sense, thank you!